les/clientpool.go

```diff
@@ -254,24 +288,21 @@ func (f *clientPool) connect(peer clientPeer, capacity uint64) bool {
 		}
 		// accept new client, drop old ones
 		for _, c := range kickList {
-			f.dropClient(c, now, true)
+			f.dropClient(c, f.clock.Now(), true)
 		}
 	}
-	// client accepted, finish setting it up
-	if nb != nil {
-		delete(f.negBalanceMap, freeID)
-		f.negBalanceQueue.Remove(nb.queueIndex)
-	}
-	if e.priority {
-		e.balanceTracker.addCallback(balanceCallbackZero, 0, func() { f.balanceExhausted(id) })
-	}
+	// Register new client to connection queue.
 	f.connectedMap[id] = e
 	f.connectedQueue.Push(e)
-	f.connectedCapacity += e.capacity
-	totalConnectedGauge.Update(int64(f.connectedCapacity))
+	f.connectedCap += e.capacity
+	// If the current client is a paid client, notify it to update the capacity.
+	// And also monitor the status of client, downgrade it to normal client if
+	// positive balance is used up.
 	if e.capacity != f.freeClientCap {
 		e.peer.updateCapacity(e.capacity)
+		e.balanceTracker.addCallback(balanceCallbackZero, 0, func() { f.balanceExhausted(id) })
```

Comment:
Hmm, if the capacity of paid client is even equal with a free client, why the client needs to pay?

I think we should have a strict rule that if the client is paid, then the capacity should larger than the free one.

Reply:
I disagree. I do not know yet what the usual practices of the market are going to be but paying for a minimal capacity definitely does make sense after you have used up your free service allowance for the near future. Actually, if you need a larger capacity and you know many good servers then it does make sense to buy the minimal capacity from multiple servers instead of buying a large capacity connection from a single one. Sure, if some of them gives you that minimal capacity for free then why not use it for free, but that is not going to be the case for a very long time.
So we should not tie priority to larger than minimum capacity. Priority means positive balance.

Reply:
Thanks for the explanation. Now I get the point.